spectator-ext-placeholders/src/main/java/com/netflix/spectator/placeholders/PlaceholderId.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spectator.placeholders;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Tag;
+
+import java.util.Map;
+
+/**
+ * An extension of the {@link Id} interface that allows the list of tag names attached
+ * to the Id to be declared in advance of the use of the metric.  This can be used to
+ * provide a default value for a tag or to use a TagFactory implementation that uses
+ * context available in the execution environment to compute the value of the tag.
+ */
+public interface PlaceholderId {
+  /** Description of the measurement that is being collected. */
+  String name();
+
+  /** Other dimensions that can be used to classify the measurement. */
+  Iterable<Tag> tags();
```

Comment:
Given that this interface no longer extends Id, I don't think that this method should be here.  People should be using the resolveToId method instead.